Parser/Token.cs

```diff
@@ -38,9 +39,20 @@ public enum E {
    // Utility function used to echo an error to the console
    public void PrintError () {
       if (Kind != ERROR) throw new Exception ("PrintError called on a non-error token");
-      Console.ForegroundColor = ConsoleColor.Yellow;
-      Console.WriteLine ($"At line {Line}, column {Column} of {Source.FileName}: {Text}");
-      Console.ResetColor ();
+      string s = $"File: {Source.FileName}", msg = $"{s}\r\n{new string ('\u2500', s.Length)}\r\n";
+      var (idx, n, space) = (Line - 2, Source.Lines.Length, new string (" "));
+      for (int i = idx; i <= Line; i++, idx++)
+         if (idx >= 1) msg += $"{idx,4}\u2502{space}{Source.Lines[idx - 1]}\r\n";
+      Write (msg);
+      ForegroundColor = ConsoleColor.Yellow; var nSpaces = Column + 5;
+      CursorLeft = nSpaces;
+      Write ("^\r\n");
+      CursorLeft = Math.Max (0, (nSpaces - Text.Length / 2) + 1);

```

Comment:
What about the case that the Text.Length causes the text to go to the right beyond Console.WindowWidth? Can you put another clamp on CursorLeft to avoid that possibility. For eample, WindowWidth = 80, Column = 70, Text.Length = 40...